disperser/encoder/server_v2.go

```diff
@@ -224,11 +223,18 @@ func (s *EncoderServerV2) validateAndParseRequest(req *pb.EncodeBlobRequest) (co
 	if req.EncodingParams.ChunkLength == 0 {
 		return blobKey, params, status.Error(codes.InvalidArgument, "chunk length must be greater than zero")
 	}
+	if req.EncodingParams.ChunkLength&(req.EncodingParams.ChunkLength-1) != 0 {
+		return blobKey, params, status.Error(codes.InvalidArgument, "chunk length must be power of 2")
+	}
 
 	if req.EncodingParams.NumChunks == 0 {
 		return blobKey, params, status.Error(codes.InvalidArgument, "number of chunks must be greater than zero")
 	}
 
+	if req.BlobSize == 0 || uint64(encoding.GetBlobLength(uint(req.BlobSize))) > req.EncodingParams.ChunkLength*req.EncodingParams.NumChunks {
+		return blobKey, params, status.Error(codes.InvalidArgument, "blob size is invalid")
```

Comment:
Seems like all of these return invalid argument, should we just wrap the upper error instead?

Reply:
yea. I'm thinking a proposal to improve error handling across the codebase. It's currently vague, leaky (like this rate limiting error in Encoder, or wrong (like the above GetBlobStatus); these should be addressed comprehensively.